packages/create-app/templates/template-tutorial-todo-aip-app/src/useProjects.ts

```diff
@@ -0,0 +1,69 @@
+import { useCallback } from "react";
+import useSWR from "swr";
+import type { MockProject } from "./mocks";
+import Mocks from "./mocks";
+
+function useProjects() {
+  const { data, isLoading, isValidating, error, mutate } = useSWR<
+    MockProject[]
+  >("projects", async () => {
+    // Try to implement this with the Ontology SDK!
+    return Mocks.getProjects();
+  });
+
+  const createProject: (
+    name: string,
+  ) => Promise<MockProject["$primaryKey"]> = useCallback(
+    async (name: string) => {
+      // Try to implement this with the Ontology SDK!
+      const id = await Mocks.createProject({ name });
+      await mutate();
+      return id;
+    },
+    [mutate],
+  );
+
+  const getRecommendedProjectDescription: (
+    project: MockProject,
+  ) => Promise<string> = useCallback(
+    async (project) => {
+      // Try to implement this with the Ontology SDK!
+      return Mocks.getRecommendedProjectDescription(project);
+    },
+    [],
+  );
+
+  const updateProjectDescription: (
+    project: MockProject,
+  ) => Promise<void> = useCallback(
+    async (project) => {
+      // Try to implement this with the Ontology SDK!
+      const description = await Mocks.getRecommendedProjectDescription(project);
+      await Mocks.updateProjectDescription(project, description);
+      await mutate();
+    },
+    [mutate],
+  );
+
+  const deleteProject: (project: MockProject) => Promise<void> = useCallback(
+    async (project) => {
+      // Try to implement this with the Ontology SDK!
+      await Mocks.deleteProject(project.$primaryKey);
+      await mutate();
+    },
+    [mutate],
+  );
+
+  return {
+    projects: data,
+    isLoading,
+    isValidating,
+    isError: error,
+    createProject,
+    deleteProject,
+    getRecommendedProjectDescription,
```

Comment:
No need to export this as it is not used by any consumers

Reply:
IsError is used in `createProjectButton`

Reply:
I was referring to `getRecommendedProjectDescription` because we directly call updateProjectDescription

Reply:
Did we try to remove `getRecommendedProjectDescription` from the return object? I don't think it is being used by any consumer.